includes/class-wc-privacy.php

```diff
@@ -178,533 +223,92 @@ public static function anonymize_completed_orders( $limit = 20, $page = 1 ) {
-					'name'  => $name,
-					'value' => $value,
-				);
-			}
-		}
-
-		/**
-		 * Allow extensions to register their own personal data for this order for the export.
-		 *
-		 * @since 3.4.0
-		 * @param array    $personal_data Array of name value pairs to expose in the export.
-		 * @param WC_Order $order An order object.
-		 */
-		$personal_data = apply_filters( 'woocommerce_privacy_export_personal_data_order', $personal_data, $order );
-
-		return $personal_data;
-	}
-
-	/**
-	 * Anonymize/remove personal data for a given EMAIL ADDRESS. This user may not have an account.
-	 *
-	 * Note; this is separate to account deletion. WooCommerce handles account deletion/cleanup elsewhere.
-	 * This logic is simply to clean up data for guest users.
-	 *
-	 * @todo Add option to determine if order data should be left alone when removing personal data for a user.
-	 * @todo Hook into core UI.
-	 *
-	 * @param string $email_address Customer email address.
+	 * @param int $timestamp Timestamp to delete customers before.
+	 * @param int $limit     Limit number of users to delete per run.
+	 * @return int Count of customers that were deleted.
 	 */
-	public static function remove_personal_data( $email_address ) {
-		$user        = get_user_by( 'email', $email_address ); // Check if user has an ID in the DB.
-		$has_account = $user instanceof WP_User;
-
-		/**
-		 * Allow 3rd parties to modify this behavior. If true, orders belonging to this user will be anonyimized.
-		 *
-		 * @since 3.4.0
-		 */
-		if ( apply_filters( 'woocommerce_privacy_remove_personal_data_includes_orders', true, $email_address ) ) {
-			$order_query = array(
-				'limit' => -1,
-			);
-
-			if ( $user instanceof WP_User ) {
-				$order_query['customer_id'] = (int) $user->ID;
-			} else {
-				$order_query['billing_email'] = $email_address;
-			}
-
-			$orders = wc_get_orders( $order_query );
-
-			if ( 0 < count( $orders ) ) {
-				foreach ( $orders as $order ) {
-					self::remove_order_personal_data( $order );
-				}
-			}
-		}
-
-		// Revoke download permissions.
-		$data_store = WC_Data_Store::load( 'customer-download' );
-
-		if ( $user instanceof WP_User ) {
-			$data_store->delete_by_user_id( (int) $user->ID );
-		} else {
-			$data_store->delete_by_user_email( $email_address );
-		}
+	protected static function delete_inactive_accounts_query( $timestamp, $limit = 20 ) {
+		$count      = 0;
+		$user_query = new WP_User_Query( array(
+			'fields'     => 'ID',
+			'number'     => $limit,
+			'role__in'   => apply_filters( 'woocommerce_delete_inactive_account_roles', array(
+				'Customer',
+				'Subscriber',
+			) ),
+			'meta_query' => array(
+				'relation' => 'AND',
+				array(
+					'key'     => 'wc_last_active',
+					'value'   => (string) $timestamp,
+					'compare' => '<',
+					'type'    => 'NUMERIC',
+				),
+				array(
+					'key'     => 'wc_last_active',
+					'value'   => '0',
+					'compare' => '>',
+					'type'    => 'NUMERIC',
+				),
+			),
+		) );
```

Comment:
If a user never logs in after a site is updated to 3.4 they will never get cleaned up by this. Just a heads-up so you're aware. 

I think the best solution if that issue needs to be addressed is to have the update script give all users a `wc_last_active` of the current time when the update script runs.

Reply:
Good idea. Done this now.